openhtf/plugs/__init__.py

```diff
@@ -407,6 +421,12 @@ def initialize_plugs(self, plug_types=None):
       self.update_plug(plug_type, plug_instance)
     self._initialize_rpc_server()
 
+  def call_trigger(self, trigger_info):
+    """Call the trigger corresponding to trigger_info on the matching plug."""
+    trigger = getattr(self._plugs_by_type[trigger_info.plug_type],
```

Comment:
you might consider catching and reraising here with a more specific error message than AttributeError will have.